week2/starter/ColorPicker/ContentView.swift

```diff
@@ -39,48 +39,56 @@ struct ContentView: View {
   @State private var blueColor: Double = 0.0
   @State private var foregroundColor = Color(red: 0, green: 0, blue: 0)
 
-  var body: some View {
+  @Environment(\.verticalSizeClass) var verticalSizeClass
+  @Environment(\.horizontalSizeClass) var horizontalSizeClass
 
-    VStack {
-      Text("Color Picker")
-        .font(.largeTitle)
+  var body: some View {
+    let verticalOrientation = verticalSizeClass == .regular && horizontalSizeClass == .compact
+    let layout = verticalOrientation ? AnyLayout(VStackLayout(spacing: Constants.General.spacing)) : AnyLayout(HStackLayout(spacing: Constants.General.spacing))
 
-      RoundedRectangle(cornerRadius: 0)
-        .foregroundColor(foregroundColor)
-        .border(.black)
-      VStack {
-        Text("Red")
-        HStack {
-          Slider(value: $redColor, in: 0...255)
-          Text("\(Int(redColor.rounded()))")
-        }
-      }
-      VStack {
-        Text("Green")
-        HStack {
-          Slider(value: $greenColor, in: 0...255)
-          Text("\(Int(greenColor.rounded()))")
+    ZStack {
+      Color("BackgroundColor")
+        .ignoresSafeArea()
+      layout {
+        VStack {
+          Text("Color Picker")
+            .font(.largeTitle)
+            .fontWeight(.bold)
+          PickedColor(color: $foregroundColor)
+            .aspectRatio(
+              verticalOrientation ?
+              Constants.General.verticalAspectRatio :
+                Constants.General.horizontalAspectRatio,
+              contentMode: .fit
+            )
         }
-      }
-      VStack {
-        Text("Blue")
-        HStack {
-          Slider(value: $blueColor, in: 0...255)
-          Text("\(Int(blueColor.rounded()))")
+
+        VStack {
+          ColorSlider(title: "Red", accentColor: .red, value: $redColor)
+          ColorSlider(title: "Green", accentColor: .green, value: $greenColor)
+          ColorSlider(title: "Blue", accentColor: .blue, value: $blueColor)
+
+          SetButton(title: "Set Color", action: setColor)
         }
       }
-      Button("Set Color") {
-        foregroundColor = Color(red: redColor / 255, green: greenColor / 255, blue: blueColor / 255)
-      }
+      .padding(Constants.General.padding)
     }
-    .background(Color.white)
-    .padding(20)
+  }
 
+  func setColor() {
```

Comment:
I like how you extracted the Button action into a function which you passed in as a property to the SetButton struct.